models/biswap/bnb/biswap_v3_bnb_trades.sql

```diff
@@ -0,0 +1,115 @@
+{{ config(
+    tags=['dunesql'],
+    schema = 'biswap_v3_bnb',
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["bnb"]\',
+                                "project",
+                                "biswap_v3",
+                                \'["chef_seaweed"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-06-26' %}
+
+
+WITH dexs AS
+(
+    --Biswap v3 on BNB Chain
+    SELECT
+        t.evt_block_time AS block_time
+        ,CAST(NULL AS VARBINARY)as taker
+        ,CAST(NULL AS VARBINARY)as maker
+        ,CASE WHEN sellXEarnY = true THEN abs(amountY) ELSE abs(amountX) END AS token_bought_amount_raw
+        ,CASE WHEN sellXEarnY = true THEN abs(amountX) ELSE abs(amountY) END AS token_sold_amount_raw
+        ,NULL AS amount_usd
+        ,CASE WHEN sellXEarnY = true THEN tokenY ELSE tokenX END AS token_bought_address
+        ,CASE WHEN sellXEarnY = true THEN tokenX ELSE tokenY END AS token_sold_address
+        ,t.contract_address as project_contract_address
+        ,t.evt_tx_hash AS tx_hash
+        ,CAST(NULL as array<bigint>) AS trace_address
```

Comment:
we are moving forward with removing this field from `dex.trades` lineage, as it's not used in any models (only aggregators). you can look at uniswap / sushiswap / pancakeswap spells which have been migrated as reference.

plz do remove this field in all spells here and the schema files. also includes unique keys, unique combo of columns test, etc.

Reply:
this applies to other dex as well